dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterPipeline.java

```diff
@@ -24,7 +24,7 @@
  */
 public class DynoDualWriterPipeline extends DynoJedisPipeline {
     private static final org.slf4j.Logger logger = LoggerFactory.getLogger(DynoDualWriterPipeline.class);
-    private static ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+    private static ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
Any performance implications because of this? Or were we never supposed to use a ThreadPool in the first place?

Is it possible to "join" on the threads (or is there some mechanism to wait on the tasks running on the threads to complete) as a pre-condition to the sync() ?

Reply:
We were not supposed to use ThreadPool.